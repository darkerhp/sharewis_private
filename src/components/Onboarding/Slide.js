import React  from 'react';
import PropTypes from 'prop-types';
import ReactNative from 'react-native';
import Hyperlink from 'react-native-hyperlink';
import I18n from 'react-native-i18n';

import BaseStyles from '../../lib/baseStyles';
import { ACT_SITE_URL } from '../../lib/constants';
import redirectTo from '../../utils/linking';

const {
  StyleSheet,
  Text,
  Image,
  View,
} = ReactNative;

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  slideImageContainer: {
    flex: 2,
    backgroundColor: BaseStyles.onboardingBackgroundColor,
  },
  slideImage: {
    flex: -1,
    justifyContent: 'center',
    marginTop: 90,
    marginBottom: 30,
    width: BaseStyles.deviceWidth,
  },
  content: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: 25,
    opacity: 0.6,
  },
  contentText: {
    textAlign: 'center',
    lineHeight: 22,
    fontSize: 17,
    color: '#222',
  },
});


const Slide = ({ text, imageSrc }) =>
  (
    <View style={styles.slide}>
      <View style={styles.slideImageContainer}>
        <Image
          source={imageSrc}
          style={styles.slideImage}
          resizeMode={'contain'}
        />
      </View>
      <View style={styles.content}>
        <Hyperlink
          style={{ flex: 1 }}
          linkStyle={{ color: BaseStyles.hyperlink }}
          linkText={url => (url === ACT_SITE_URL ? I18n.t('actWebsite') : url)}
          onPress={redirectTo}
        >

          <Text style={styles.contentText}>
            {text}
          </Text>
        </Hyperlink>
      </View>
    </View>
  );

Slide.propTypes = {
  text: PropTypes.string.isRequired,
  imageSrc: PropTypes.number.isRequired,
};

export default Slide;
