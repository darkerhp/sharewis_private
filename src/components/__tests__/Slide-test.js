/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import Slide from '../Onboarding/Slide';

jest.mock('../../utils/linking', () => () => 'redirectTo');
jest.mock('react-native-i18n', () => ({ currentLocale: () => '' }));

describe('Slide', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Slide />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
