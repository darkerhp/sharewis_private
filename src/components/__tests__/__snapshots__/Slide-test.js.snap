exports[`Slide renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#ecf1f3",
        "flex": 2,
      }
    }>
    <Image
      resizeMode="contain"
      source={undefined}
      style={
        Object {
          "flex": -1,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 90,
          "width": 375,
        }
      } />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "opacity": 0.6,
        "padding": 25,
      }
    }>
    <View
      linkStyle={
        Object {
          "color": "#2980b9",
        }
      }
      linkText={[Function]}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontSize": 17,
            "lineHeight": 22,
            "textAlign": "center",
          }
        } />
    </View>
  </View>
</View>
`;
